src/org/zaproxy/zap/extension/fuzz/payloads/generator/JsonPayloadGenerator.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2015 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.fuzz.payloads.generator;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.natpryce.snodge.JsonMutator;
+import org.zaproxy.zap.extension.fuzz.payloads.DefaultPayload;
+import org.zaproxy.zap.utils.ResettableAutoCloseableIterator;
+
+import java.util.Iterator;
+
+public class JsonPayloadGenerator implements StringPayloadGenerator {
+    private final String json;
+    private final int numberOfPayloads;
+    private JsonElement originalJson;
```

Comment:
Making this final would be better, how about changing `assertIsValid` to `createElement` which returns the `JsonElement`? It could then be assigned to `originalJson` in the constructor.